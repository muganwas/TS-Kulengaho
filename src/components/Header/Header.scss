@import '../../styles/colors.scss';
@import '../../styles/font-sizes.scss';
.header-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: $lightGrey;
  padding: 5px;
  .logo-title-container {
    display: flex;
    flex-direction: row;
    align-items: center;
    .header-text {
      font-weight: bold;
      font-size: $page-title;
    }
    .logo {
      width: 40px;
      height: auto;
      margin: 5px;
    }
  }
  .sub-title {
    font-size: $page-subtitle;
    text-transform: lowercase;
    font-style: italic;
  }
}
