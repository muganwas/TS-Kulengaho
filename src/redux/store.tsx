import { combineReducers, applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { apiMiddleware } from 'redux-api-middleware';
import logger from 'redux-logger';

var middleware;
// apply logger only in development and staging
if (process.env.NODE_ENV !== 'production') middleware = applyMiddleware(promise, thunk, apiMiddleware, logger);
else middleware = applyMiddleware(promise, thunk, apiMiddleware);

const initialState = {};

const allReducers = combineReducers({
//put all reducers here
});

const store = createStore(
    allReducers,
    initialState,
    middleware 
)

export default store;