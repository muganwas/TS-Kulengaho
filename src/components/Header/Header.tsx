import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { useHistory } from 'react-router-dom'
import { setUserLogInStatus } from '../../redux/actions'
import './Header.scss'
import { RootState } from '../../sharedTypes'
import logo from '../../images/logo.svg'
import { Button } from '../Form'

interface headerProps {
	title: String
}

const Header: React.FC<headerProps> = ({ title }) => {
	const dispatch = useDispatch()
	const history = useHistory()
	const subTitle = useSelector(
		(store: RootState) => store.genInfo.site_subtitle
	)
	const userInfo = useSelector((store: RootState) => store.userInfo)
	const logout: () => void = () => {
		dispatch(setUserLogInStatus(false))
		history.push('/')
	}
	return (
		<div data-testid='header-container' className='header-container'>
			<div className='logo-title-container'>
				<img
					data-testid='logo'
					alt='logo'
					className='logo'
					src={logo}
				/>
				<div className='header-text'>{title}</div>
			</div>
			<div className='sub-title'>{subTitle}</div>
			{userInfo.is_logged_in && (
				<div id='logout-container'>
					<Button
						id={'logout-button'}
						testid='logout-button'
						containerClass='logout-button-container'
						buttonClass='logout-button'
						title='Logout'
						onClick={logout}
					/>
				</div>
			)}
		</div>
	)
}

export default Header
