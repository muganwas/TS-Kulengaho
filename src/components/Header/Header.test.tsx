import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from '../../redux/store'
import Header from './Header'

describe('Header component renders', () => {
	it('snapshot checks out', () => {
		render(
			<Provider store={store}>
				<Header title='Kulengaho' />
			</Provider>
		)
		const container = screen.getAllByTestId('header-container')
		expect(container).toMatchSnapshot()
	})
})
