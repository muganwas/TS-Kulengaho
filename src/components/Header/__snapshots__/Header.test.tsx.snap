// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders snapshot checks out 1`] = `
Array [
  <div
    class="header-container"
    data-testid="header-container"
  >
    <div
      class="logo-title-container"
    >
      <img
        alt="logo"
        class="logo"
        data-testid="logo"
        src="logo.svg"
      />
      <div
        class="header-text"
      >
        Kulengaho
      </div>
    </div>
    <div
      class="sub-title"
    >
      Kulengaho
    </div>
  </div>,
]
`;
